### PR TITLE
Lowercase heading ids

### PR DESCRIPTION
**Pull request by emplums**
@simurai pointed out that the Table of Contents links were broken in primer/css because our `MarkdownHeading` component wasn't creating lower case id's, which remark/toc expects :) 

Bumps version to 4.0.2

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@primer/blueprints",
-  "version": "4.0.0",
```
Comment (emplums) [inline package.json]: Why does it say the version here is 4.0.0, but when I go into the Code tab it says the version is 4.0.1 in this repo 🤔

Comment (emplums) [reply package.json]: https://github.com/primer/blueprints/blob/master/package.json

Comment (shawnbot) [reply package.json]: I think it's because of the merge conflict? I thought that if there's a conflict we showed the diff between the PR branch and the parent SHA of the branch point, but [that commit](https://github.com/primer/blueprints/commit/f93db8aef3ae1f60054651626c43a415a3821ca8) is the `v4.0.1` tag, so I'm kinda stumped. 🤔

Comment (shawnbot) [reply package.json]: Huh yeah, resolving the conflict "fixed" the diff. 🤷‍♀

File: src/MarkdownHeading.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ const StyledHeading = styled(Heading)`
   }
 `
 const MarkdownHeading = ({children, className, ...rest}) => {
-  const id = slugify(children.toString())
+  const id = slugify(children.toString(), {lower: true})
```
Comment (shawnbot) [inline src/MarkdownHeading.js]: Do we assert anywhere that `children` is `toString()`-able? 😬

Comment (emplums) [reply src/MarkdownHeading.js]: Hmm no, I wonder what would happen if the child was a component 🤔 I'll test it out

Comment (emplums) [reply src/MarkdownHeading.js]: image
😬

Comment (emplums) [reply src/MarkdownHeading.js]: I'm not really sure if it's worth putting in provisions for edge cases in the code, we can just add to the documentation that the children should be text only. Maybe add a prop type declaration for `children`?

Comment (emplums) [reply src/MarkdownHeading.js]: Especially since this is to be used in MD or MDX files only, the only way you'd get a non-string in there is if you pass in a React component which I feel like is fair to say we should do 😬 Not sure if Markdown will even convert `### <Text>asklfjasdf</Text>` to a heading anyways

Comment (emplums) [reply src/MarkdownHeading.js]: Added in a check to make sure children isn't empty!

Comment (shawnbot) [reply src/MarkdownHeading.js]: Cool, thanks. I think in the future this is something we should look at handling with a remark plugin, since this won't work with anything _except_ a single child that is a string.
